pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -188,7 +188,9 @@ func separateVSHostsAndServices(virtualService config.Config,
 				// This service doesn't have a virtualService that matches it.
 				continue
 			}
-			foundSvcMatch = true // we did find a match
+			// If this service host matches the virtual service host at all, then it's a match
+			// and we shouldn't add it to the list of hosts to prevent duplicates
+			foundSvcMatch = svcHost.Matches(hostname)
```

Comment:
Should this be ANDed with the existing value of foundSvcMatch so it doesn’t reset an already true value from a previous iteration of the loop?